### PR TITLE
Small MMD experiment edits

### PR DESCRIPTION
**Pull request by emersodb**
# PR Type
Other

Small edits to the experiments. Also adding peft to the main dependencies because it's in the core utils

Also separating out the peft extraction functionality from the main extraction due to deepspeed being brought in. Even if you're not using the peft extraction, it gets imported because it's in the same utils file, which causes issues for all our examples on GPU if the CUDA_HOME env variable is not set.

# Tests Added

N/A

### CODE REVIEW

**Comment by emersodb** [comment]
Note: I had to inject some `type: ignores` due to this issue with transformers: https://github.com/huggingface/transformers/issues/37339

File: research/rxrx1/ditto_deep_mmd/client.py [hunk 1]
```diff
@@ -179,17 +179,11 @@ def get_model(self, config: Config) -> nn.Module:
...
-    post_aggregation_last_checkpoint_name = f"post_aggregation_client_{args.client_number}_last_model.pkl"
     checkpoint_and_state_module = ClientCheckpointAndStateModule(
         pre_aggregation=[
             BestLossTorchModuleCheckpointer(checkpoint_dir, pre_aggregation_best_checkpoint_name),
             LatestTorchModuleCheckpointer(checkpoint_dir, pre_aggregation_last_checkpoint_name),
         ],
-        post_aggregation=[
-            BestLossTorchModuleCheckpointer(checkpoint_dir, post_aggregation_best_checkpoint_name),
-            LatestTorchModuleCheckpointer(checkpoint_dir, post_aggregation_last_checkpoint_name),
-        ],
```
Comment (sanaAyrml) [inline research/rxrx1/ditto_deep_mmd/client.py:192]: Here, you’ve only removed the extra checkpointing from deep_mmd on RxRx1—should we apply this change across all models?

**Comment by emersodb** [comment]
> One thing I suggest adding to x.slrm files of cifar10 is: `export CUBLAS_WORKSPACE_CONFIG=:4096:8` Otherwise, it results in an error caused by setting the random seed.

Does it? I don't think I experienced this error when I was running the experiments.

File: research/rxrx1/ditto_deep_mmd/client.py (hunk 1 above)
Comment (emersodb) [reply research/rxrx1/ditto_deep_mmd/client.py:192]: Sure. I can apply that across the board.

**Comment by sanaAyrml** [comment]
Oh I always get it :(

**Review by sanaAyrml — APPROVED** [review]
LGTM! Thanks for edits.
